[Swift in WebKit] The Swift implementation of WKTextAnimationManager is incorrect

rr-codes opened the pull request:
#### 8af0ee64b30d69cbbb4986f2f0f899d6e8140bd5
<pre>
[Swift in WebKit] The Swift implementation of WKTextAnimationManager is incorrect
<a href="https://bugs.webkit.org/show_bug.cgi?id=294532">https://bugs.webkit.org/show_bug.cgi?id=294532</a>
<a href="https://rdar.apple.com/153476595">rdar://153476595</a>

Reviewed by Aditya Keerthi.

Fix the implementation of WKTextAnimationManager to properly use `@objc @implementation.`

* Source/WebKit/SourcesCocoa.txt:
* Source/WebKit/UIProcess/Cocoa/WebKitSwiftSoftLink.h:
* Source/WebKit/UIProcess/Cocoa/WebKitSwiftSoftLink.mm:
* Source/WebKit/UIProcess/ios/WKContentViewInteraction.h:
* Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm:
(-[WKContentView addTextAnimationForAnimationID:withData:]):
* Source/WebKit/UIProcess/mac/WebViewImpl.h:
* Source/WebKit/UIProcess/mac/WebViewImpl.mm:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/WebKitSwift/WebKitSwift.h:

- Fix naming inconsistencies; previously, there was &quot;WKSTextAnimationManager&quot;, &quot;TextAnimationManager&quot;,
and &quot;WKTextAnimationManager&quot;; the latter two were the same class, and only for iOS, and the latter
was only for macOS. Resolve this inconsistency by having the class be `WKTextAnimationManager` for all
platforms, and have platform-specific header files. [1]

* Source/WebKit/UIProcess/API/Cocoa/WKWebView.mm:
* Source/WebKit/UIProcess/API/ios/WKWebViewIOS.mm:
* Source/WebKit/UIProcess/API/mac/WKWebViewMac.mm:
* Source/WebKit/UIProcess/Cocoa/PageClientImplCocoa.mm:
* Source/WebKit/UIProcess/Cocoa/WKTextAnimationType.h: Removed.

Remove WKTextAnimationType from WebKit, since it is only needed on iOS and defined in WebKitSwift.

* Source/WebKit/UIProcess/WKSTextAnimationManager.h: Removed.

This and the delgate header file were moved into one header file for consistency.

* Source/WebKit/UIProcess/mac/WKTextAnimationManagerMac.h: Renamed from Source/WebKit/UIProcess/mac/WKTextAnimationManager.h.
* Source/WebKit/UIProcess/mac/WKTextAnimationManagerMac.mm: Renamed from Source/WebKit/UIProcess/mac/WKTextAnimationManager.mm.

This should only be compiled on macOS.

* Source/WebKit/WebKitSwift/TextAnimation/TextAnimationManager.swift: Removed.

Renamed to WKTextAnimationManagerIOS

* Source/WebKit/WebKitSwift/TextAnimation/WKTextAnimationManagerIOS.swift: Added.

- Use `@objc @implementation`
- Use feature flags
- Apply consistent formatting and style

* Source/WebKit/WebKitSwift/TextAnimation/WKTextAnimationManagerIOS.h: Renamed from Source/WebKit/WebKitSwift/TextAnimation/WKSTextAnimationSourceDelegate.h.

Canonical link: <a href="https://commits.webkit.org/296469@main">https://commits.webkit.org/296469@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/162b060e22b90688d9568a11ce1e62034db43a8b

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/108609 "1 style error") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/28270 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/18694 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/113817 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/58999 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/28959 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/36825 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/82499 "Passed tests") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/111557 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/22989 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/97831 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/62936 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/22408 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/58531 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/92358 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/16019 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/116938 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/35663 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/26297 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/91520 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/36036 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/94100 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/91324 "Found 3 new API test failures: /WebKitGTK/TestResources:/webkit/WebKitWebResource/get-data-error, /WebKitGTK/TestWebProcessExtensions:/webkit/WebKitWebView/web-process-crashed, /WebKitGTK/TestWebKitPolicyClient:/webkit/WebKitPolicyClient/new-window-policy (failure)") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/23269 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/36221 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/13983 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/31479 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/35564 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/41097 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/35274 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/38620 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/36952 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

pxlcoder reviewed (approved):
Please make sure to manually test animations to ensure some code was not lost during this refactoring.

rr-codes committed:
[Swift in WebKit] The Swift implementation of WKTextAnimationManager is incorrect
https://bugs.webkit.org/show_bug.cgi?id=294532
rdar://153476595

Reviewed by Aditya Keerthi.

Fix the implementation of WKTextAnimationManager to properly use `@objc @implementation.`

* Source/WebKit/SourcesCocoa.txt:
* Source/WebKit/UIProcess/Cocoa/WebKitSwiftSoftLink.h:
* Source/WebKit/UIProcess/Cocoa/WebKitSwiftSoftLink.mm:
* Source/WebKit/UIProcess/ios/WKContentViewInteraction.h:
* Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm:
(-[WKContentView addTextAnimationForAnimationID:withData:]):
* Source/WebKit/UIProcess/mac/WebViewImpl.h:
* Source/WebKit/UIProcess/mac/WebViewImpl.mm:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/WebKitSwift/WebKitSwift.h:

- Fix naming inconsistencies; previously, there was "WKSTextAnimationManager", "TextAnimationManager",
and "WKTextAnimationManager"; the latter two were the same class, and only for iOS, and the latter
was only for macOS. Resolve this inconsistency by having the class be `WKTextAnimationManager` for all
platforms, and have platform-specific header files. [1]

* Source/WebKit/UIProcess/API/Cocoa/WKWebView.mm:
* Source/WebKit/UIProcess/API/ios/WKWebViewIOS.mm:
* Source/WebKit/UIProcess/API/mac/WKWebViewMac.mm:
* Source/WebKit/UIProcess/Cocoa/PageClientImplCocoa.mm:
* Source/WebKit/UIProcess/Cocoa/WKTextAnimationType.h: Removed.

Remove WKTextAnimationType from WebKit, since it is only needed on iOS and defined in WebKitSwift.

* Source/WebKit/UIProcess/WKSTextAnimationManager.h: Removed.

This and the delgate header file were moved into one header file for consistency.

* Source/WebKit/UIProcess/mac/WKTextAnimationManagerMac.h: Renamed from Source/WebKit/UIProcess/mac/WKTextAnimationManager.h.
* Source/WebKit/UIProcess/mac/WKTextAnimationManagerMac.mm: Renamed from Source/WebKit/UIProcess/mac/WKTextAnimationManager.mm.

This should only be compiled on macOS.

* Source/WebKit/WebKitSwift/TextAnimation/TextAnimationManager.swift: Removed.

Renamed to WKTextAnimationManagerIOS

* Source/WebKit/WebKitSwift/TextAnimation/WKTextAnimationManagerIOS.swift: Added.

- Use `@objc @implementation`
- Use feature flags
- Apply consistent formatting and style

* Source/WebKit/WebKitSwift/TextAnimation/WKTextAnimationManagerIOS.h: Renamed from Source/WebKit/WebKitSwift/TextAnimation/WKSTextAnimationSourceDelegate.h.

Canonical link: https://commits.webkit.org/296469@main
